service/src/main/java/uk/gov/hmcts/reform/fpl/service/StandardDirectionsService.java

```diff
@@ -37,6 +44,19 @@
             .collect(toList());
     }
 
+    public boolean hasEmptyDates(Map<String, Object> data) {
+        return Stream.of(DirectionAssignee.values()).anyMatch(directionAssignee -> {
```

Comment:
Also instead of sending the map here, thoughts of using `CaseData` here and then using `Stream.of(caseData.getAllParties,...)` might be a bit longer but it will mean that there is less need for mapping inside the actual method itself.

Reply:
nice idea, updated